source/slang/slang-ir-explicit-global-context.cpp

```diff
@@ -462,11 +462,15 @@ struct IntroduceExplicitGlobalContextPass
         for( IRUse* use = globalVar->firstUse; use; use = nextUse )
         {
             nextUse = use->nextUse;
+            auto user = use->getUser();
+
+            // Ensure the use site checked actually requires a replacement
+            if (as<IRDecoration>(user))
+                continue;
```

Comment:
Shouldn't you limit this to `kIROp_GlobalVariableShadowingGlobalParameterDecoration`?
Is it safe to apply to all `IRDecoration`?

Reply:
I thought about doing that but it looks like that would be incorrect with the function logic.

The function is trying to look at uses and crashes if a use is not a child of a function. This does not make sense since IRDecoration's are not often a child of a function (if used to store auxiliary data the Decoration should not be blindly modified).